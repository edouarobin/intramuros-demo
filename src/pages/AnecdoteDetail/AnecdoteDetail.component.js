// @flow

import React, { PureComponent } from 'react';
import {
  ImageBackground,
  StyleSheet,
  View,
  Text,
  ScrollView,
  StatusBar,
  TouchableOpacity,
} from 'react-native';
// $FlowFixMe
import LinearGradient from 'react-native-linear-gradient';

import {
  GoBackButtonWithGradient,
  ImageZoomModal,
  ContactWeb,
} from 'Intramuros/src/components';
import globalStyle, {
  anecdoteDetailImageHeight,
  STATUS_BAR_HEIGHT,
  BIG_SCREEN_PADDING_OFFSET,
} from 'Intramuros/src/style/globalStyle';
import HTMLText from 'Intramuros/src/utils/HTMLText';

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    flexGrow: 1,
  },
  image: {
    width: '100%',
    height: anecdoteDetailImageHeight + STATUS_BAR_HEIGHT,
    justifyContent: 'flex-end',
  },
  titleStyle: {
    fontFamily: 'Lato-Bold',
    fontSize: 20,
    // fontWeight: 'bold',
    color: 'white',
    lineHeight: 28,
    marginBottom: 20,
    paddingHorizontal: 16,
    backgroundColor: 'transparent',
  },
  descriptionContainer: {
    backgroundColor: 'white',
    flex: 1,
    padding: 16,
    paddingHorizontal: 16 + BIG_SCREEN_PADDING_OFFSET,
  },
  decriptionStyle: {
    fontFamily: 'Lato-Regular',
    fontSize: 14,
    lineHeight: 24,
    color: 'black',
  },
  linearGradient: {
    position: 'absolute',
    flex: 1,
    zIndex: 1,
    width: '100%',
    height: '40%',
    justifyContent: 'flex-end',
  },
  linksStyle: {
    marginTop: 16,
  },
  moreInfo: {
    fontFamily: 'Lato-BoldItalic',
    color: globalStyle.colors.greyishBrown,
    marginTop: 10,
    marginBottom: 5,
  },
});

type PropsType = {
  navigation: NavigationPropsType<>,
  anecdoteDetails: AnecdoteType,
};

export default class AnecdoteDetail extends PureComponent<PropsType> {
  state = {
    modalVisible: false,
  };

  openZoomImage = () => {
    this.setState({ modalVisible: true });
  };
  closeImageModal = () => {
    this.setState({ modalVisible: false });
  };

  render() {
    return (
      <View style={styles.container}>
        <GoBackButtonWithGradient
          navigation={this.props.navigation}
          gradientColor="rgba(0,0,0,0.9)"
          goBackColor="white"
        />
        <StatusBar
          barStyle="light-content"
          translucent
          backgroundColor="transparent"
          animated={true}
        />
        <ScrollView contentContainerStyle={styles.container}>
          <TouchableOpacity onPress={this.openZoomImage}>
            <ImageBackground
              style={styles.image}
              source={{ uri: this.props.anecdoteDetails.image }}
            >
              <LinearGradient
                colors={['rgba(256,256,256,0)', 'rgba(0,0,0,0.9)']}
                style={styles.linearGradient}
              >
                <Text
                  numberOfLines={3}
                  ellipsizeMode={'tail'}
                  style={styles.titleStyle}
                >
                  {this.props.anecdoteDetails.title}
                </Text>
              </LinearGradient>
            </ImageBackground>
          </TouchableOpacity>
          <View style={styles.descriptionContainer}>
            <HTMLText
              style={styles.decriptionStyle}
              value={this.props.anecdoteDetails.description}
            />
            {this.props.anecdoteDetails.url1 ||
            this.props.anecdoteDetails.url2 ||
            this.props.anecdoteDetails.url3 ? (
              <View style={styles.linksStyle}>
                <Text style={styles.moreInfo}>Plus d'informations :</Text>
                {this.props.anecdoteDetails.url1 ? (
                  <ContactWeb
                    website={this.props.anecdoteDetails.url1}
                    fisrtElement={true}
                  />
                ) : null}
                {this.props.anecdoteDetails.url2 ? (
                  <ContactWeb website={this.props.anecdoteDetails.url2} />
                ) : null}
                {this.props.anecdoteDetails.url3 ? (
                  <ContactWeb website={this.props.anecdoteDetails.url3} />
                ) : null}
              </View>
            ) : null}
          </View>
        </ScrollView>
        <ImageZoomModal
          image={this.props.anecdoteDetails.image}
          modalVisible={this.state.modalVisible}
          closeImageModal={this.closeImageModal}
        />
      </View>
    );
  }
}
