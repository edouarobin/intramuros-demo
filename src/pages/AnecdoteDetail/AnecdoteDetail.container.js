// @flow
import { connect } from 'react-redux';

import { selectors } from 'Intramuros/src/redux/anecdotes';

import AnecdoteDetail from './AnecdoteDetail.component';

const mapStateToProps = (rootState: RootStateType, ownProps: any) => ({
  anecdoteDetails: selectors.selectAnecdoteDetailsFromID(
    rootState,
    ownProps.navigation.state.params.anecdoteID
  ),
});

const mapDispatchToProps = {};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AnecdoteDetail);
