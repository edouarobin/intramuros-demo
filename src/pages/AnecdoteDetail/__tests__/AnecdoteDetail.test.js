// @flow-weak

import React from 'react';
import renderer from 'react-test-renderer';

import AnecdoteDetail from '../AnecdoteDetail.component';

const mockAnecdoteDetails = {
  title: 'Recette : La soupe angevine',
  description: `Découvrez ce cocktail angevin qui perdrait de son charme si on remplaçait le cointreau par de la  liqueur. On peut remplacer le crémant par du mousseux, ce qui est moins coûteux. 
  
  Tout d’abord, vous devez vous munir des ingrédients suivants :
  
       1 bouteille de crémant de Loire
       1 citron
       15 cl de cointreau
       12 cl de sirop de sucre de Cannes
  
  Tout en laissant la bouteille de crémant de Loire bien au frais, commencez par presser le citron
  Découvrez ce cocktail angevin qui perdrait de son charme si on remplaçait le cointreau par de la  liqueur. On peut remplacer le crémant par du mousseux, ce qui est moins coûteux. 
  
  Tout d’abord, vous devez vous munir des ingrédients suivants :
  
       1 bouteille de crémant de Loire
       1 citron
       15 cl de cointreau
       12 cl de sirop de sucre de Cannes`,
  createdAt: '2017-11-10T15:06:03.295894Z',
  updatedAt: '2017-11-28T17:09:19.978928Z',
  image:
    'https://elasticbeanstalk-eu-west-2-309758705895.s3.amazonaws.com/anecdotes/6cba01e87dab75af0bf4e7b662212ce2_Recette___D%C3%A9couvrez_.jpg',
  id: 1,
};

it('Fully renders the AnecdoteDetails page', () => {
  const tree = renderer
    .create(
      <AnecdoteDetail navigation={{}} anecdoteDetails={mockAnecdoteDetails} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
