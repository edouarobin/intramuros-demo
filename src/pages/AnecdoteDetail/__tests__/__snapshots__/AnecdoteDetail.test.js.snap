// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fully renders the AnecdoteDetails page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flexGrow": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        3858759680,
        16777215,
      ]
    }
    end={undefined}
    locations={null}
    start={undefined}
    style={
      Object {
        "flex": 1,
        "height": 320.16,
        "position": "absolute",
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 8,
          "paddingTop": 30,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": 100,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "width": 100,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 36,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "black",
                    "marginHorizontal": 8,
                  },
                  Object {
                    "color": "white",
                  },
                ],
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "black",
                  "fontSize": 16,
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Retour
          </Text>
        </View>
      </View>
    </View>
  </BVLinearGradient>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "white",
        "flexGrow": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "height": 622.3000000000001,
            "justifyContent": "flex-end",
            "width": "100%",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://elasticbeanstalk-eu-west-2-309758705895.s3.amazonaws.com/anecdotes/6cba01e87dab75af0bf4e7b662212ce2_Recette___D%C3%A9couvrez_.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 622.3000000000001,
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <BVLinearGradient
          colors={
            Array [
              16777215,
              3858759680,
            ]
          }
          end={undefined}
          locations={null}
          start={undefined}
          style={
            Object {
              "flex": 1,
              "height": "40%",
              "justifyContent": "flex-end",
              "position": "absolute",
              "width": "100%",
              "zIndex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 20,
                "fontWeight": "bold",
                "lineHeight": 28,
                "marginBottom": 20,
                "paddingHorizontal": 16,
              }
            }
          >
            Recette : La soupe angevine
          </Text>
        </BVLinearGradient>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "padding": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 14,
              "lineHeight": 24,
            }
          }
        >
          Découvrez ce cocktail angevin qui perdrait de son charme si on remplaçait le cointreau par de la  liqueur. On peut remplacer le crémant par du mousseux, ce qui est moins coûteux. 
  
  Tout d’abord, vous devez vous munir des ingrédients suivants :
  
       1 bouteille de crémant de Loire
       1 citron
       15 cl de cointreau
       12 cl de sirop de sucre de Cannes
  
  Tout en laissant la bouteille de crémant de Loire bien au frais, commencez par presser le citron
  Découvrez ce cocktail angevin qui perdrait de son charme si on remplaçait le cointreau par de la  liqueur. On peut remplacer le crémant par du mousseux, ce qui est moins coûteux. 
  
  Tout d’abord, vous devez vous munir des ingrédients suivants :
  
       1 bouteille de crémant de Loire
       1 citron
       15 cl de cointreau
       12 cl de sirop de sucre de Cannes
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
